lib/eventq/eventq_aws/sns.rb

```diff
@@ -21,7 +21,7 @@ def create_topic_arn(event_type, region = nil)
         _event_type = EventQ.create_event_type(event_type)
         topic_key = "#{region}:#{_event_type}"
 
-        arn = get_topic_arn(event_type, region)
+        arn = @@topic_arns[topic_key]
         unless arn
           response = sns.create_topic(name: aws_safe_name(_event_type))
```

Comment:
Really great spot, that should improve performance as lot :)
> Creates a topic to which notifications can be published. Users can create at most 100,000 standard topics (at most 1,000 FIFO topics). For more information, see [Creating an Amazon SNS topic](https://docs.aws.amazon.com/sns/latest/dg/sns-create-topic.html) in the Amazon SNS Developer Guide . This action is idempotent, so if the requester already owns a topic with the specified name, that topic's ARN is returned without creating a new topic.

Reply:
Thanks! I'm still worried someone was depending on weird behaviour, so I'll cut a major version just to be safe. I like to believe there was a reason for the previous approach... 😁